src/main/kotlin/de/studiocode/miniatureblocks/builderworld/BuilderWorld.kt

```diff
@@ -94,7 +94,7 @@ class BuilderWorld : Listener {
     private fun Location.isValidBuildArea(builderWorld: BuilderWorld): Boolean {
         val chunk = this.chunk
 
-        return world == builderWorld.world && chunk.x % 2 == 0 && chunk.z % 2 == 0 && y > 0 && y < 17
+        return world == builderWorld.world && chunk.x and 1 != 0 && chunk.z and 1 != 0 && y > 0 && y < 17 // If the least significant bit is set the number is not even.
```

Comment:
I'm not quite sure about that one, it makes the code harder to read and the performance benefit is probably negligible.

Reply:
**Note that the reviewed file is outdated, so don't test this. Test the current version**

The comment should help the issue of hard readability. And the performance does indeed benefit a lot from this. While a modern compiler would translate this automatically, kotlin bytecode is still performing quite poorly and wouldn't do this automatically. Here's a quick example:

I want to know if 57 is even.

### Remainder / Modulo

- First, I have to divide 57 by 2 and convert it to an integer. => `57 / 2 = 28.5` => `(Int) 28.5 = 28`
- Then take that integer and multiply it with 2 => `28 * 2 = 56`
- Then I have to take the resulting integer and subtract it from 57 => `57 - 56 = 1`.
- And finally, compare that to 0 => `1 == 0 = false`.

But what's important here is the division since it gets harder to compute the larger the number gets.

### Bitwise

(Note that this is only a byte since I didn't want to include all the unnecessary zeros)

* First, compute the bitwise and of 57 and 1 => `0b00111001 & 0b00000001 = 0b00000001`
* Then compare it to 0 => `1 == 0 = false`

Just from looking at these steps, you can already see the complexity reducing by a lot